sql/src/main/java/io/crate/operation/projectors/ShardRequestAccumulator.java

```diff
@@ -95,8 +108,13 @@ public int batchSize() {
         if (currentRequest.itemIndices().isEmpty() && isLastBatch) {
             return CompletableFuture.completedFuture(getSingleRowWithRowCount());
         }
+
+        final String nodeId = getLocalNodeId();
```

Comment:
I thought of keeping it close to its usage, but don't have a strong opinion about it. I genuinely see no benefits/drawbacks either way.